src/blocks/scratch3_sensing.js

```diff
@@ -217,7 +220,19 @@ class Scratch3SensingBlocks {
 
     getLoudness () {
         if (typeof this.runtime.audioEngine === 'undefined') return -1;
-        return this.runtime.audioEngine.getLoudness();
+        if (typeof this.runtime.currentStepTime === 'undefined') return -1;
+
+        // Only measure loudness once per step
+        if (this.loudnessTime) {
```

Comment:
Once you initialize loudnessTime in the constructor, you won't need this outer `if`, because you can initialize loudness time to zero, and timer.time() - 0 will always be larger than the step time on the first step.

Reply:
all set